### PR TITLE
Test transaction and use as async context manager

### PR DESCRIPTION
**Pull request by cybergrind**
1. Add test for Transaction
2. Make it usable as an async context manager

### CODE REVIEW

**Comment by cybergrind** [comment]
hey @rhdxmr. would you like and have time to do cross-reviews?

**Comment by rhdxmr** [comment]
@cybergrind Sure.

File: aiozk/transaction.py [hunk 1]
```diff
@@ -68,6 +69,15 @@ async def commit(self):
...
         return result
 
+    async def __aenter__(self):
+        return self
+
+    async def __aexit__(self, ext_type, exception, tb):
+        result = await self.commit()
+        if not result:
+            raise TransactionFailed
+        return result
```
Comment (rhdxmr) [inline aiozk/transaction.py:79]: ``ext_type`` seems misspelled. And ``return`` statement is redundant.

What about adding ``if`` branch for the case of ``ext_type`` is not None? That means an exception occurs inside ``async with`` statement.

Comment (cybergrind) [reply aiozk/transaction.py]: @rhdxmr yeah, all makes sense. fixed both

**Comment by rhdxmr** [comment]
@cybergrind Looks good. I am merging this PR. Thank you

**Comment by rhdxmr** [comment]
I mistakenly made a merge commit.

**Comment by cybergrind** [comment]
@rhdxmr no problem. next time just use rebase without merge-commit, please

**Comment by rhdxmr** [comment]
OK I got it

**Comment by cybergrind** [comment]
There is a rule that can enforce that, so next time it won't allow you to push without having a linear history :) thus we won't have such issue anyway.
I've rebased master back, so you will need to do `git pull -r` in your local master if you did `git pull` before.

**Pull request merged**